Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1505,21 +1509,58 @@ function createAnimatedComponent(Component: any): any {
 
     componentWillUnmount() {
       this._propsAnimated && this._propsAnimated.__detach();
+      this._detachNativeEvents(this.props);
     }
 
     setNativeProps(props) {
       this._component.setNativeProps(props);
     }
 
     componentWillMount() {
-      this.attachProps(this.props);
+      this._attachProps(this.props);
     }
 
     componentDidMount() {
       this._propsAnimated.setNativeView(this._component);
+
+      this._attachNativeEvents(this.props);
+    }
+
+    _attachNativeEvents(newProps) {
+      if (newProps !== this.props) {
```

Comment:
We should perhaps add to the documentation that creating event in render method is not the most effective way